### DISCUSSION TITLE
Ideas overview

### DISCUSSION THREAD

**Discussion by lucaszischka** [LucasMucGH/BottomSheet#68] [Ideas]
# This is a collection of ideas that need to be discussed.

Each idea is briefly explained and my opinion is given.
Any feedback for further ideas or comments on the ideas (whether they are redundant, great, ...) are welcomed.

## Fixing shadow (#87, #88)
- Currently the shadow is still visible for a short time when the BottomSheet hides.

> **My thoughts on this:**
> This is definitely needed.
> However I currently neither have the motivation (after working more than one month on updating to version 3) nor the time to make this large of a change to the project.
> Feel free to add this feature via pull request yourself (see Solution).

#### Solution:
- Replace the `.move(.bottom)` transitions with `.offset(y: _)`
- Optional: *Cleanup height calculations to make offset easier to calculate*
- Add a `.customBackgroundShadow` modifier
- Add the shadow height to the offset transition y value

## Making the background modifiable individually (#87, #88)
- For example:`.customBackgroundShadow`, `.customBackgroundMaterial`, `.customBackgroundCornerRadius`, ...

> **My thoughts on this:**
> The `.customBackground(...)` modifier can be used to change everything about the background.
> I understand that it you may want to change only one thing and not replace the whole background, but I think there are way to many "one things" people may want to change, so that I will end up with dozens of different view modifiers one day.

## Making the close button more flexible (see #66, #69)
- custom action to replace the one which closes the sheet
- custom icon
 
> **My thoughts on this:**
> The close button is a relict of the early phases of the BottomSheet. I think there are many use cases for it, so I didn't removed it jet.
> But you can still simply use completely custom header, and place any button on the left without compromises.
> So I think it would be far beyond our objective to make it customisable. 

## Make the BottomSheetPosition protocol based (like apples implementation)

> **My thoughts on this:**
> I think apple did a great job on solving the height/BottomSheetPosition issues.
> I will need further investigation, if apples way to implement this is compatible with mine (especially regarding dynamic height).
> But if it is, I will change the BottomSheetPosition to make it protocol based.

### Important:
- No ETA is and will be provided.
- Please note that this discussion will be updated frequently, so if some comments refer to things that are not on the list, they are either already implemented or scrapped.

**Comment by TariqAlmazyad** [comment]
Beautiful work .

**Comment by NathHorrigan** [comment]
Hey @LucasMucGH  - Any idea on when we might see functionality for `appleScrollBehavior`?

**Comment by lucaszischka** [comment]
Hey @NathHorrigan,
as long as i don’t find a way to prevent a ScrollView from scrolling i can’t tell you a estimated arrival time.
Greetings Lucas

**Comment by blandinf** [comment]
It will be possible with the new update to update topCornerRadius and background blur color? 
Thanks for your work!

**Comment by lucaszischka** [comment]
> It will be possible with the new update to update topCornerRadius and background blur color?

Yes, I put it on my list. Greetings Lucas

**Comment by dewit** [comment]
@LucasMucGH Hi Lucas, love this package, thanks for creating it. Could you share a timeframe for some of the decided features in this list? It would greatly enhance the app I'm working on. Thanks in advance. Keep up the good work :)

**Comment by lucaszischka** [comment]
Hello Arjan,

im currently very busy and don’t have much time going into expanding the project.
I wont share a ETA because this is a project I do for fun in my free time and I don’t want to stress myself, by setting deadlines.
As soon as I have more time again I will continue expanding this project and I will add more features, but for the time being, the best option if you want to see new features is to make a pull request and adding them yourself. Sorry about that.

Greetings,
Lucas

**Comment by dewit** [comment]
> Hello Arjan,
> 
> im currently very busy and don’t have much time going into expanding the project. I wont share a ETA because this is a project I do for fun in my free time and I don’t want to stress myself, by setting deadlines. As soon as I have more time again I will continue expanding this project and I will add more features, but for the time being, the best option if you want to see new features is to make a pull request and adding them yourself. Sorry about that.
> 
> Greetings, Lucas

I get that completely! No problem at all. Thanks anyway :)

**Comment by Sherlouk** [comment]
Hey @LucasMucGH 👋  Just wanted to say what a lovely library this is. I've been exploring a few alternatives (mostly including Apple's own new one) and this has so many awesome features in comparison to theirs 👏 

Was going to raise a ticket but realised somebody raised it before me, so would love to +1 on number 15 (pull down the sheet instead of scrolling the ScrollView when at the top). I understand you're not spending too much time on this at the moment 🙂  I may look to open a PR (time permitting on my side too aha!).

Awesome work regardless ❤️

**Comment by flowbe** [comment]
I'd love to see the dynamic height feature! Hope it's implemented soon 🙏🏻
Awesome work nonetheless!

**Comment by rouuuge** [comment]
First: Awesome work!!

Is there already a way to check if the sheet is visible and the animation ended?

**Reply by lucaszischka** [reply]
The bottom sheet should be visible in every position, except hidden. Also in bottomPosition only the header content is visible.
